python/cudf/cudf/core/indexed_frame.py

```diff
@@ -699,6 +699,35 @@ def drop_duplicates(
             self._index.names if not ignore_index else None,
         )
 
+    @annotate("FRAME_EMPTY_LIKE", color="green", domain="cudf_python")
+    def _empty_like(self, keep_index=True):
+        return self._from_columns_like_self(
+            libcudf.copying.columns_empty_like(
+                [
+                    *(self._index._data.columns if keep_index else ()),
+                    *self._columns,
+                ]
+            ),
+            self._column_names,
+            self._index.names if keep_index else None,
+        )
+
+    def _split(self, splits, keep_index=True):
+        return [
+            self._from_columns_like_self(
+                libcudf.copying.columns_split(
+                    [
+                        *(self._index._data.columns if keep_index else []),
+                        *self._columns,
+                    ],
+                    splits,
+                )[split_idx],
```

Comment:
Unless I miss something, this is duplicating work right? We call the same `columns_split` every time in the loop, then just subset with `split_idx`. If we run into issues with referencing the same columns multiple times, could we just copy?

Reply:
Yes, this is duplicating work :man_facepalming: